onnxruntime/core/providers/trt/trt_execution_provider.cc

```diff
@@ -252,8 +252,9 @@ common::Status TRTExecutionProvider::Compile(const std::vector<onnxruntime::Node
       }
     }
 
-    for (auto& i : output_to_add) {
-      *(model_proto.mutable_graph()->mutable_output()->Add()) = graph_value_info[i];
+    auto* output_add = model_proto.mutable_graph()->mutable_output()->Add();
```

Comment:
Are you sure this is correct? The call to Add() would add an output only once if you do it outside the loop this way and *output_add would keep overwriting the same memory location.

Reply:
corrected. Thanks.

Reply:
Is there a unit test that should have caught this issue?

---
In reply to: [261488608](https://github.com/Microsoft/onnxruntime/pull/382#discussion_r261488608) [](ancestors = 261488608)

Reply:
Unit tests for TRT is WIP because issues are found in onnx-tensorrt parser and need to be addressed first.